### PR TITLE
Fetch auth token automatically if not specified in API test

### PR DESCRIPTION
**Pull request by jayasg12**
CLUSTER_TYPE=$1

function get_bearer_token() {
        if [[ $CLUSTER_TYPE == "openshift" ]]; then
                echo "Fetching AUTH_TOKEN for Openshift"
                AUTH_TOKEN=$(oc whoami --show-token)
        else
                echo "Fetching AUTH_TOKEN for Cloudctl"
                AUTH_TOKEN=$(cloudctl tokens | grep "Bearer"  | cut -d' ' -f5)
        fi
}

get_bearer_token

### CODE REVIEW

**Comment by ShishirH** [comment]
Thanks for the contribution @jayasg12 . A few changes:
1. When the user passes the cluster_type (oc or icp), we should first check if the corresponding tools (oc and cloudctl) exist, and if not report an error and exit, similar to how we are doing for jq.
2. Because of https://github.com/kruize/kruize/pull/106, we have additional parameters like URL, port etc. Due to this, we should not be using the first parameter as the cluster type, but should be using -c, ie, `./testAPI -u <URL> -c <CLUSTER_TYPE>`. This can be done in the same way as the above PR does, using getopts.
3. Right now `get_bearer_token` is always being called. It should only be called when the user specifies the cluster_type, otherwise, it should continue with default values.

**Comment by dinogun** [comment]
Please rebase and fix conflicts. Thanks

File: tests/testAPI.sh [hunk 1]
```diff
@@ -41,6 +44,16 @@ function check_auth_token() {
...
 }
 
+function get_bearer_token() {
+        if [[ $CLUSTER_TYPE == "openshift" ]]; then
+		check_tool_existance "oc"
+                AUTH_TOKEN=$(oc whoami --show-token)
+        elif [[ $CLUSTER_TYPE == "ICP" ]]; then
+                check_tool_existance "cloudctl"
+                AUTH_TOKEN=$(cloudctl tokens | grep "Bearer"  | cut -d' ' -f5)
+        fi
```
Comment (dinogun) [inline tests/testAPI.sh:54]: What happens if the cluster type is neither openshift nor ICP ? Please use a validation function to ensure that the CLUSTER_TYPE is valid

File: tests/testAPI.sh [hunk 2]
```diff
@@ -62,11 +75,25 @@ do
...
 		[?])
 			usage
 	esac
 done
 
+check_tool_existance "jq"
+
+if [[ $AUTH_TOKEN == "" ]]; then
+	if [[ -z $CLUSTER_TYPE ]]; then
+		echo "Error - Enter the Cluster Type !!"
```
Comment (dinogun) [inline tests/testAPI.sh:90]: Here you have ask the user to either enter the AUTH_TOKEN or set the CLUSTER_TYPE.

File: tests/testAPI.sh [hunk 3]
```diff
@@ -62,11 +75,25 @@ do
...
+                exit 1 
+	else  
+		get_bearer_token ${CLUSTER_TYPE}
+        fi
```
Comment (dinogun) [inline tests/testAPI.sh:94]: Please use tabs everywhere, this line appears to be misaligned.

**Comment by jayasg12** [comment]
@dinogun changes requested are done and Pushed. Please verify and merge this PR. Thanks !!

**Pull request merged**